### PR TITLE
Feature/mms 88

### PR DESCRIPTION
**Pull request by HuiJun**

### CODE REVIEW

File: crud/src/main/java/org/openmbee/sdvc/crud/services/DefaultProjectService.java [hunk 1]
```diff
@@ -90,6 +103,8 @@ public ProjectJson update(ProjectJson project) {
                 }
             }
             projectRepository.save(proj);
+            projectIndex.update(project);
```
Comment (dlamoris) [inline crud/src/main/java/org/openmbee/sdvc/crud/services/DefaultProjectService.java]: don't think project has docId set here

File: crud/src/main/java/org/openmbee/sdvc/crud/controllers/projects/ProjectsController.java [hunk 2]
```diff
@@ -54,7 +58,12 @@ public ProjectsResponse getAllProjects(
             if (mss.hasProjectPrivilege(auth, proj.getProjectId(), Privileges.PROJECT_READ.name(), true)) {
                 ProjectJson projectJson = new ProjectJson();
                 projectJson.merge(convertToMap(proj));
-                response.getProjects().add(projectJson);
+                Optional<ProjectJson> projectJsonOption = projectIndex.findById(proj.getDocId());
+                if (projectJsonOption.isPresent()) {
+                    response.getProjects().add(projectJsonOption.get());
+                } else {
+                    response.getProjects().add(projectJson);
+                }
```
Comment (dlamoris) [inline crud/src/main/java/org/openmbee/sdvc/crud/controllers/projects/ProjectsController.java]: need to update single project get below

File: elastic/src/main/java/org/openmbee/sdvc/elastic/ProjectElasticImpl.java [hunk 3]
```diff
@@ -87,4 +111,20 @@ private String read(InputStream in) throws IOException {
...
+            request.docAsUpsert(true).doc(json).upsert(json);
+            client.update(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    protected String getIndex() {
+        return super.getIndex() + "_metadata";
```
Comment (dlamoris) [inline elastic/src/main/java/org/openmbee/sdvc/elastic/ProjectElasticImpl.java:125]: super.getIndex depends on context projectid being set, check if the project endpoints set that before getting here

File: elastic/src/main/java/org/openmbee/sdvc/elastic/ProjectElasticImpl.java [hunk 4]
```diff
@@ -38,13 +46,23 @@ public void create(String projectId, String projectType) {
             commitIndex.mapping(getCommitMapping(), XContentType.JSON);
             CreateIndexRequest nodeIndex = new CreateIndexRequest(index + "_node");
             nodeIndex.mapping(getNodeMapping(projectType), XContentType.JSON);
+            CreateIndexRequest metadataIndex = new CreateIndexRequest(index + "_metadata");
+            nodeIndex.mapping(getMetadataMapping(), XContentType.JSON);
```
Comment (dlamoris) [inline elastic/src/main/java/org/openmbee/sdvc/elastic/ProjectElasticImpl.java]: should be metadataIndex

Comment (dlamoris) [reply elastic/src/main/java/org/openmbee/sdvc/elastic/ProjectElasticImpl.java]: also i don't see the metadata mapping commited?

**Review by dlamoris — CHANGES_REQUESTED** [review]

File: crud/src/main/java/org/openmbee/sdvc/crud/services/DefaultProjectService.java (hunk 1 above)
Comment (HuiJun) [reply crud/src/main/java/org/openmbee/sdvc/crud/services/DefaultProjectService.java]: should be set on line 75

File: elastic/src/main/java/org/openmbee/sdvc/elastic/ProjectElasticImpl.java (hunk 4 above)
Comment (HuiJun) [reply elastic/src/main/java/org/openmbee/sdvc/elastic/ProjectElasticImpl.java]: outdated

File: crud/src/main/java/org/openmbee/sdvc/crud/services/DefaultProjectService.java (hunk 1 above)
Comment (dlamoris) [reply crud/src/main/java/org/openmbee/sdvc/crud/services/DefaultProjectService.java]: that's in the create function though?

**Pull request merged**